src/main/scala/edu/ie3/simona/sim/SimonaSim.scala

```diff
@@ -236,19 +237,30 @@ class SimonaSim(simonaSetup: SimonaSetup)
   }
 
   def stopAllChildrenGracefully(
+      simulationSuccessful: Boolean,
       listenerDelay: FiniteDuration = 500.millis
   ): Unit = {
     gridAgents.foreach { case (gridAgentRef, _) =>
       context.unwatch(gridAgentRef)
     }
-    gridAgents.foreach(_._1 ! Finish)
+    gridAgents.foreach(_._1 ! StopMessage(simulationSuccessful))
 
     context.unwatch(scheduler)
     context.stop(scheduler)
 
     context.unwatch(weatherService)
     context.stop(weatherService)
 
+    extSimulationData.allActorsAndInitTriggers.foreach { case (ref, _) =>
+      context.unwatch(ref)
+    }
```

Comment:
From my point of view, it is a bit more comprehensible, if you unwatch the refs within the two `foreach` in the following lines. It seems, that you operate on three collections, wheres it's actually only two.

Reply:
Good point